### PR TITLE
Draft : Add Support to pimcore 11

### PR DESCRIPTION
**Pull request by ogmdev**
Working on Pimcore 11 support

### Done
- [x] Bump to SF 5.4
- [x] Bump to PHP8.1

### TODO (not exhaustive)

- [x] Bump to SF 6.2
- [x] Bump to PHP8.2
- [ ] Upgrade docs
- [ ]  Fix CSP errors
- [x] Remove Pimcore\Extension\Bundle\Traits\StateHelperTrait

### CODE REVIEW

**Comment by matthiashamacher** [comment]
@ogmdev When should this be released? Many of our projects cannot be updated as of this incompatibility.

**Comment by ctippler** [comment]
Hi @ogmdev , 
first of all thank you very much for your work to bumping it to Pimcore 11. We really appreciate it :-).
Is it currently working with Pimcore 11. I guess there will be some smaller issues, but if it is basically working i would suggest that we merge it to the master so that others can contribute with further bugfixes...

What do you think?

Regards,
Christian

**Comment by ogmdev** [comment]
Hi @ctippler 

Nice idea ! 
Yesterday afternoon, I made some tests on a professional project which be updated to Pimcore 11.
Everything seems okay.

For CSP errors, I have, for the moment, disabled the headers
```yaml
pimcore_admin:
    admin_csp_header:
        enabled: false
```

Olivier

File: src/ExecutionTrait.php [hunk 1]
```diff
@@ -256,42 +252,44 @@ public static function checkExecutingUser($allowedUsers = [])
...
 
-            for($x = 1; $x <= 3; $x++){
-                $result = Helper::executeJob($monitoringItem->getConfigurationId(), $monitoringItem->getCallbackSettings(), 0, json_encode($package), $monitoringItem->getId(), $callback);
+            for($x = 1; $x <= 3; $x++) {
+                $result = Helper::executeJob($monitoringItem->getConfigurationId(), $monitoringItem->getCallbackSettings(), 0, json_encode($package, JSON_THROW_ON_ERROR), $monitoringItem->getId(), $callback);
 
-                if($result['success'] == false){
-                    $monitoringItem->getLogger()->warning("Can't start child (tried ". $i ." time ) - reason: " . $result['message']);
+                if($result['success'] == false) {
+                    $monitoringItem->getLogger()->warning("Can't start child (tried ". $i .' time ) - reason: ' . $result['message']);
```
Comment (NiklasBr) [inline src/ExecutionTrait.php:287]:
```suggestion
                    $attempts = $i === 1 ? "$i time" : "$i times";
                    $monitoringItem->getLogger()->warning("Can't start child (tried $attempts) - reason: " . $result['message']);


```

**Comment by kvaka-dev** [comment]
I get an error when installing bundle on clean project without any previous configuration

`Elements\Bundle\ProcessManagerBundle\Model\Configuration::getById(): Argument #1 ($id) must be of type string|int, null given, called in /var/www/html/vendor/elements/process-manager-bundle/src/SystemEventsListener.php on line 66`

@ctippler I suppose you upgraded on existing configurations?

**Comment by ctippler** [comment]
hi @kvaka-dev , this should be fixed in the latest master version.

Regards,
Christian

**Comment by kvaka-dev** [comment]
@ctippler yes, it is fixed. thanks.

**Pull request merged**